metagraph/src/graph/graph_extensions/node_rc.hpp

```diff
@@ -0,0 +1,61 @@
+#ifndef __NODE_RC_HPP__
+#define __NODE_RC_HPP__
+
+#include <sdsl/dac_vector.hpp>
+#include <sdsl/rrr_vector.hpp>
+
+#include "graph/representation/base/sequence_graph.hpp"
+#include "common/vectors/bit_vector_adaptive.hpp"
+
+namespace mtg {
+namespace graph {
+
+class DBGSuccinct;
+
+// Maps each node in a PRIMARY-mode DeBruijnGraph to nodes adjacent to its reverse
+// complement. When using the index construct constructor on DBGSuccinct, or when
+// loading such an index, this stores a map from each DBGSuccinct to the BOSS nodes
+// corresponding to the reverse complements of its k-1 prefix and k-1 suffix.
+template <class Indicator = bit_vector_smart,
+          class Mapping = sdsl::dac_vector_dp<sdsl::rrr_vector<>>>
+class NodeRC : public SequenceGraph::GraphExtension {
```

Comment:
I don't see how having these flexible template types helps. You're not going to change them. Pick the best option and always use that.

Reply:
I still want to keep this flexibility in case I decide to play around with different compression schemes, but it'll probably be useful to rename this to something like TNodeRC and make NodeRC a typedef with these types